File: architeture-sample/basiclib/src/main/java/example/basiclib/util/ObjectUtils.java
package example.basiclib.util;

/**
 * Email yummyl.lau@gmail.com
 * Created by yummylau on 2018/01/25.
 */
public class ObjectUtils {

    public static boolean equals(Object o1, Object o2) {
        if (o1 == null) {
            return o2 == null;
        }
        if (o2 == null) {
            return false;
        }
        return o1.equals(o2);
    }
}

File: libBase/src/main/java/com/effective/android/base/util/ObjectUtils.kt
package com.effective.android.base.util

/**
 * Email yummyl.lau@gmail.com
 * Created by yummylau on 2018/01/25.
 */
object ObjectUtils {

    @JvmStatic
    fun equals(o1: Any?, o2: Any?): Boolean {
        if (o1 == null) {
            return o2 == null
        }
        return if (o2 == null) {
            false
        } else o1 == o2
    }
}
